standalone/node/src/service.rs

```diff
@@ -253,12 +253,17 @@ pub fn new_partial(
 		let select_chain = select_chain.clone();
 		let keystore = keystore_container.sync_keystore();
 		let chain_spec = config.chain_spec.cloned_box();
-		let backend = backend.clone();
-		let is_archive_mode = match config.state_pruning {
-			PruningMode::Constrained(_) => false,
-			PruningMode::ArchiveAll | PruningMode::ArchiveCanonical => true,
-		};
 
+		let rpc_backend = backend.clone();
+		let is_archive_mode = match &config.state_pruning {
+			Some(m) => {
+				match m {
+					PruningMode::Constrained(_) => false,
+					PruningMode::ArchiveAll | PruningMode::ArchiveCanonical => true,
+				}
+			},
+			None => false
```

Comment:
I guess it would not prune when the `state_pruning` is None. So we should treat it as `is_archive_mode=true`.
Then we have not to add `--pruning=archive` in the e2e script.

Reply:
I'll try

Reply:
You're right